flink-streaming-java/src/test/java/org/apache/flink/streaming/api/datastream/ReinterpretDataStreamAsKeyedStreamITCase.java

```diff
@@ -227,5 +293,22 @@ public void close() throws Exception {
 			Assert.assertEquals(expectedSum, runningSum);
 			super.close();
 		}
+
+		@Override
+		public void snapshotState(FunctionSnapshotContext context) throws Exception {
+			sumState.add(runningSum);
+		}
+
+		@Override
+		public void initializeState(FunctionInitializationContext context) throws Exception {
+			sumState = context.getOperatorStateStore().getListState(
+				new ListStateDescriptor<>("sumState", Integer.class));
+
+			if (context.isRestored()) {
```

Comment:
Where do we check whether or not we actually have restore something?

Reply:
Through through the restore flag and a validating sink? If state was dropped, the validation would go wrong.